src/handler/promote.ts

```diff
@@ -0,0 +1,100 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import 'dotenv/config';
+import { cloneDeep } from 'lodash';
+import { StatusCode } from '../models/StatusCode.enum';
+import { TechrecordGet } from '../models/post';
+import { PromoteRecordRequestBody } from '../models/promote';
+import { SearchCriteria } from '../models/search';
+import { getBySystemNumberAndCreatedTimestamp, searchByCriteria, updateVehicle } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { validatePromoteErrors } from '../validators/promote';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Promote end point called');
+
+  const promoteErrors = validatePromoteErrors(event);
+
+  if (promoteErrors) {
+    return addHttpHeaders(promoteErrors);
+  }
+
+  const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber ?? '');
+  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
+  const userDetails = getUserDetails(event.headers.Authorization ?? '');
```

Comment:
getUserDetail needs to be in the try block, it'll throw and error if name and id is missing

Reply:
tbh all of it needs to be? Really unsure why the try is where it is